### ISSUE TITLE
Bug report "Turning "bookmark on post" setting on makes every post on 2ch.hk a bookmark of the same thread and every single bookmark leads to 404 error screen."

### ISSUE THREAD

**Issue by K1rakishou** [Adamantcheese/Kuroba#613]
Turning "bookmark on post" setting on makes every post on 2ch.hk a bookmark of the same thread and every single bookmark leads to 404 error screen.

```
E/Kuroba | ChanThreadLoader: Thread has no op!
E/Kuroba | ChanThreadLoader: Thread has no op!
E/Kuroba | ChanThreadLoader: Thread has no op!
E/Kuroba | ChanThreadLoader: Thread has no op!
E/Kuroba | ChanThreadLoader: Thread has no op!
E/Kuroba | ChanThreadLoader: Thread has no op!
E/Kuroba | ChanThreadLoader: Thread has no op!
E/Kuroba | ChanThreadLoader: Thread has no op!
E/Kuroba | ChanThreadLoader: Thread has no op!
E/Kuroba | ChanThreadLoader: Thread has no op!
E/Kuroba | ChanThreadLoader: Thread has no op!
E/Kuroba | ChanThreadLoader: Thread has no op!
E/Kuroba | ChanThreadLoader: Thread has no op!
E/Kuroba | ChanThreadLoader: Thread has no op!
E/Kuroba | ChanThreadLoader: Thread has no op!
E/Kuroba | ChanThreadLoader: Thread has no op!
E/Kuroba | ChanThreadLoader: Thread has no op!
E/Kuroba | ChanThreadLoader: Thread has no op!
E/Kuroba | ChanThreadLoader: Thread has no op!
E/Kuroba | ChanThreadLoader: Thread has no op!
E/Kuroba | ChanThreadLoader: Thread has no op!
E/Kuroba | ChanThreadLoader: Thread has no op!
E/Kuroba | ChanThreadLoader: Thread has no op!
E/Kuroba | ChanThreadLoader: Thread has no op!
E/Kuroba | ChanThreadLoader: Thread has no op!
E/Kuroba | ChanThreadLoader: Thread has no op!
E/Kuroba | ConcurrentChunkedFileDownloader: [NormalFileCacheV2Thread-0]: Error while trying to download, class = CancellationException, message = CancellationException for request with url = https://2ch.hk/%2Fb%2Fsrc%2F213270226%2F15812390754800.png, state = Canceled
I/Kuroba | ChanThreadLoader: Loading error
I/Kuroba | ChanThreadLoader: com.android.volley.ClientError
I/Kuroba | ChanThreadLoader:    at com.android.volley.toolbox.BasicNetwork.performRequest(BasicNetwork.java:199)
I/Kuroba | ChanThreadLoader:    at com.android.volley.NetworkDispatcher.processRequest(NetworkDispatcher.java:132)
I/Kuroba | ChanThreadLoader:    at com.android.volley.NetworkDispatcher.processRequest(NetworkDispatcher.java:111)
I/Kuroba | ChanThreadLoader:    at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:90)
E/Kuroba | ReplyPresenter: onPostComplete error
I/Kuroba | ChanThreadLoader: Loading error
I/Kuroba | ChanThreadLoader: com.android.volley.ClientError
I/Kuroba | ChanThreadLoader:    at com.android.volley.toolbox.BasicNetwork.performRequest(BasicNetwork.java:199)
I/Kuroba | ChanThreadLoader:    at com.android.volley.NetworkDispatcher.processRequest(NetworkDispatcher.java:132)
I/Kuroba | ChanThreadLoader:    at com.android.volley.NetworkDispatcher.processRequest(NetworkDispatcher.java:111)
I/Kuroba | ChanThreadLoader:    at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:90)
I/Kuroba | ChanThreadLoader: Loading error
I/Kuroba | ChanThreadLoader: com.android.volley.ClientError
I/Kuroba | ChanThreadLoader:    at com.android.volley.toolbox.BasicNetwork.performRequest(BasicNetwork.java:199)
I/Kuroba | ChanThreadLoader:    at com.android.volley.NetworkDispatcher.processRequest(NetworkDispatcher.java:132)
I/Kuroba | ChanThreadLoader:    at com.android.volley.NetworkDispatcher.processRequest(NetworkDispatcher.java:111)
I/Kuroba | ChanThreadLoader:    at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:90)
```

![Screenshot_2020-02-09-15-14-35-094_com github adamantcheese chan](https://user-images.githubusercontent.com/10332458/74104498-497e9000-4b66-11ea-9fc1-e911421a4fb1.jpg)

![Screenshot_2020-02-09-15-14-37-929_com github adamantcheese chan](https://user-images.githubusercontent.com/10332458/74104509-67e48b80-4b66-11ea-81a6-014a65622120.jpg)

**Comment by K1rakishou** [comment]
Most likely related to https://github.com/Adamantcheese/Kuroba/issues/554

**Comment by Adamantcheese** [comment]
The local loadable being constructed is the same for every post probably. There are other strange things with the Dvach API anyways, like infinitely scrolling boards using index.json instead of catalog.json for the catalog.

**Comment by K1rakishou** [comment]
I think I forgot to mention, but if you boormark it normally - it works. It only doesn't work for bookmarks created via bookmark-on-post feature. So probably bookmark-on-post uses wrong op value or something like that.

**Comment by Adamantcheese** [comment]
It's likely that the localLoadable declaration is generating the same loadable with a `no` of 0, as the threadNo and postNo fields don't get updated in the DvachReplyCall process call unless there's an error. That means that for any single board, every auto-bookmarked thread should be the same, but different boards should auto-bookmark different (but still identical) posts.

**Comment by Adamantcheese** [comment]
Closed with #892 refactoring.
